src/serialization/interval_log/mod.rs

```diff
@@ -0,0 +1,667 @@
+    ///
+    /// If you do silly things like write a comment with a newline character, you'll end up with
+    /// an un-parseable log file. Don't do that.
+    pub fn add_comment(&mut self, s: &str) -> &mut Self {
+        self.comments.push(s.to_owned());
+        self
+    }
+
+    /// Set a StartTime. See the module-level documentation for more info.
+    ///
+    /// This can be called multiple times, but only the value for the most recent invocation will
+    /// be written.
+    pub fn with_start_time(&mut self, seconds_since_epoch: f64) -> &mut Self {
+        self.start_time = Some(seconds_since_epoch);
+        self
+    }
+
+    /// Set a BaseTime. See the module-level documentation for more info.
+    ///
+    /// This can be called multiple times, but only the value for the most recent invocation will
+    /// be written.
+    pub fn with_base_time(&mut self, seconds_since_epoch: f64) -> &mut Self {
+        self.base_time = Some(seconds_since_epoch);
+        self
+    }
+
+    /// Set a max value divisor.
+    ///
+    /// This is used to scale down the max value part of an interval log to something that may be
+    /// more human readable. The max value in the log is only for human consumption, so you might
+    /// prefer to divide by 10<sup>9</sup> to turn nanoseconds into fractional seconds, for
+    /// instance.
+    ///
+    /// If this is not set, 1.0 will be used.
+    ///
+    /// This can be called multiple times, but only the value for the most recent invocation will
+    /// be written.
+    pub fn with_max_value_divisor(&mut self, max_value_divisor: f64) -> &mut Self {
+        self.max_value_divisor = max_value_divisor;
+        self
+    }
+
+    /// Build a LogWriter and apply any configured headers.
+    pub fn build_with<'a, 'b, W: 'a + io::Write, S: 'b + Serializer>(
+        &self,
+        writer: &'a mut W,
+        serializer: &'b mut S,
+    ) -> Result<IntervalLogWriter<'a, 'b, W, S>, io::Error> {
+        let mut internal_writer = InternalLogWriter {
+            writer,
+            serializer,
+            text_buf: String::new(),
+            serialize_buf: Vec::new(),
+            max_value_divisor: self.max_value_divisor,
+        };
+
+        for c in self.comments.iter() {
+            internal_writer.write_comment(&c)?;
+        }
+
+        if let Some(st) = self.start_time {
+            internal_writer.write_fmt(format_args!(
+                "#[StartTime: {:.3} (seconds since epoch)]\n",
+                st
+            ))?;
+        }
+
+        if let Some(bt) = self.base_time {
+            internal_writer.write_fmt(format_args!(
+                "#[BaseTime: {:.3} (seconds since epoch)]\n",
+                bt
+            ))?;
+        }
+
+        // The Java impl doesn't write a comment for this but it's confusing to silently modify the
+        // max value without leaving a trace
+        if self.max_value_divisor != 1.0_f64 {
```

Comment:
Straying off the reservation here but it is so easy to misinterpret a log (as a human) without this

Reply:
I like it.